### PR TITLE
Timestamps refactor (farm-ng-amiga)

### PR DESCRIPTION
**Pull request by Hackerman342**

### CODE REVIEW

File: protos/farm_ng/oak/oak.proto [hunk 1]
```diff
@@ -64,11 +64,13 @@ message CameraSettings {
...
-  int64 sequence_num = 3;  // for synchronization between left, right and stereo
-  double timestamp = 4; // seconds, synchronized with host monotonic
-  double timestamp_device = 5; // seconds, device monotonic clock
+  // NOTE: timestamp that we really want is capture time synchronized to the host monotonic clock
+  int64 category = 1;          // DepthAI catetory
+  int64 instance_num = 2;      // DepthAI instance number
+  int64 sequence_num = 3;      // for synchronization between left, right and stereo
+  double timestamp = 4;        // seconds, capture time device monotonic clock
+  double timestamp_device = 5; // seconds, capture time device Clock::now
+  double timestamp_host = 7;   // seconds, host receive time in monotonic clock
```
Comment (ethanrublee) [inline protos/farm_ng/oak/oak.proto:73]: i would call this ``timestamp_recv``.

File: protos/farm_ng/oak/oak.proto [hunk 2]
```diff
@@ -64,11 +64,13 @@ message CameraSettings {
...
-  int64 category = 1; // DepthAI catetory
-  int64 instance_num = 2;  //DepthAI instance number
-  int64 sequence_num = 3;  // for synchronization between left, right and stereo
-  double timestamp = 4; // seconds, synchronized with host monotonic
-  double timestamp_device = 5; // seconds, device monotonic clock
+  // NOTE: timestamp that we really want is capture time synchronized to the host monotonic clock
+  int64 category = 1;          // DepthAI catetory
+  int64 instance_num = 2;      // DepthAI instance number
+  int64 sequence_num = 3;      // for synchronization between left, right and stereo
+  double timestamp = 4;        // seconds, capture time device monotonic clock
```
Comment (ethanrublee) [inline protos/farm_ng/oak/oak.proto:71]: The comment here is wrong.  This is the timestamp, on the host's monotonic clock, when the image was captured (the beginning of the exposure).  Note this timestamp is not monotonic as the timestamp is synchronized with the host. 

References:
https://docs.luxonis.com/projects/hardware/en/latest/pages/guides/sync_frames/#frame-capture-graphs
https://github.com/luxonis/depthai/issues/214

timestamp_device is the device's monotonic clock, beginning of the exposure and is monotonic.

File: protos/farm_ng/oak/oak.proto [hunk 3]
```diff
@@ -81,16 +83,18 @@ message OakGyro {
   Vec3F32 gyro = 1;
   int32 sequence_num = 2;
   string accuracy = 3;
-  double timestamp = 4;
-  double timestamp_device = 5;
+  double timestamp = 4;        // See OakImageMeta
+  double timestamp_device = 5; // See OakImageMeta
+  double timestamp_host = 6;   // See OakImageMeta
```
Comment (ethanrublee) [inline protos/farm_ng/oak/oak.proto:88]: timestamp_recv

File: protos/farm_ng/oak/oak.proto [hunk 4]
```diff
@@ -81,16 +83,18 @@ message OakGyro {
...
 
 message OakAccelero {
   Vec3F32 accelero = 1;
   int32 sequence_num = 2;
   string accuracy = 3;
-  double timestamp = 4;
-  double timestamp_device = 5;
+  double timestamp = 4;        // See OakImageMeta
+  double timestamp_device = 5; // See OakImageMeta
+  double timestamp_host = 6;   // See OakImageMeta
```
Comment (ethanrublee) [inline protos/farm_ng/oak/oak.proto:97]: timestamp_recv

**Review by ethanrublee — CHANGES_REQUESTED** [review]

**Review by ethanrublee — APPROVED** [review]

File: protos/farm_ng/gps/gps.proto [hunk 5]
```diff
@@ -36,9 +36,31 @@ message GpsFrameStatus {
...
+    int32 valid_time = 12;     // uint8 validTime -> Valid UTC Time of Day
+    int32 fully_resolved = 13; // uint8 fullyResolved -> UTC time of day has been fully resolved (no seconds uncertainty)
+    int32 valid_mag = 14;      // uint8 validMag -> Valid Magnetic Declination
+}
+
 message GpsFrame {
-    farm_ng.core.proto.Timestamp stamp = 1;
-    farm_ng.core.proto.Timestamp gps_time = 14;
+    farm_ng.core.proto.Timestamp stamp = 1;      // seconds, host receive time in monotonic clock
+    farm_ng.core.proto.Timestamp gps_time = 14;  // ITOW (integer time of the week) on the device
```
Comment (Hackerman342) [inline protos/farm_ng/gps/gps.proto:63]: Change comment if reference clock of `gps_time` is set to change

File: protos/farm_ng/gps/gps.proto [hunk 6]
```diff
@@ -56,11 +78,12 @@ message GpsFrame {
...
     double height = 19;             // Height above ellipsoid
     GpsFrameStatus status = 13;
+    UtcStamp utc_stamp = 20;
 }
 
 message RelativePositionFrame {
-  farm_ng.core.proto.Timestamp stamp = 1;
-  farm_ng.core.proto.Timestamp gps_time = 2;
+  farm_ng.core.proto.Timestamp stamp = 1;    // seconds, host receive time in monotonic clock
+  farm_ng.core.proto.Timestamp gps_time = 2; // ITOW (integer time of the week) on the device
```
Comment (Hackerman342) [inline protos/farm_ng/gps/gps.proto:86]: Change comment if reference clock of gps_time is set to change

**Review by edgarriba — APPROVED** [review]

File: protos/farm_ng/gps/gps.proto (hunk 5 above)
Comment (edgarriba) [reply protos/farm_ng/gps/gps.proto:63]: `stamp` in line 62 i feel we should give a more specific name

Comment (Hackerman342) [reply protos/farm_ng/gps/gps.proto:63]: Let's save some refactoring of the GPS service & redefining the protos to better match the ubx messages for its own PR

**Pull request merged**